trace-utils/src/send_data.rs

```diff
@@ -133,14 +222,76 @@ impl SendData {
             size,
             target: target.clone(),
             headers,
+            retry_strategy: RetryStrategy::default(),
+        }
+    }
+
+    /// Overrides the default RetryStrategy with user-defined values.
+    ///
+    /// # Arguments
+    ///
+    /// * `retry_strategy`: The new retry strategy to be used.
+    pub fn set_retry_strategy(&mut self, retry_strategy: RetryStrategy) {
+        self.retry_strategy = retry_strategy;
+    }
+
+    pub async fn send(self) -> SendDataResult {
+        if self.use_protobuf() {
+            self.send_with_protobuf().await
+        } else {
+            self.send_with_msgpack().await
+        }
+    }
+
+    // This function wraps send_data with a retry strategy and the building of the request.
+    // Hyper doesn't allow you to send a ref to a request, and you can't clone it. So we have to
+    // build a new one for every send attempt.
+    async fn send_payload(
+        &self,
+        content_type: &'static str,
+        payload: Vec<u8>,
+    ) -> Result<Response<Body>, SendRequestError> {
+        let mut request_attempt = 0;
+
+        loop {
+            request_attempt += 1;
+            let mut req = self.create_request_builder();
+            req = req.header("Content-type", content_type);
+
+            let result = self.send_request(req, payload.clone()).await;
```

Comment:
This should be a `Bytes` that you clone instead to share the underlying data, and the `Bytes` can be created outside the `loop` with `Bytes::from(payload)`